adcc/test_properties.py

```diff
@@ -81,3 +83,27 @@ def base_test(self, system, method, kind):
         ref = refdata[method][kind]
         n_ref = len(state.excitation_vector)
         assert_allclose(res_dms, ref["state_dipole_moments"][:n_ref], atol=1e-4)
+
+
+class TestState2StateTransitionDipoleMoments(unittest.TestCase, Runners):
+    def base_test(self, system, method, kind):
+        method = method.replace("_", "-")
+        if "cvs" in method:
+            skip("State-to-state transition dms not yet implemented for CVS.")
+
+        refdata = cache.reference_data[system]
+        state = cache.adc_states[system][method][kind]
+
+        state_to_state = refdata[method][kind]["state_to_state"]
+        refevals = refdata[method][kind]["eigenvalues"]
+        for i, exci in enumerate(state.excitations):
+            assert exci.excitation_energy == refevals[i]
+            fromi_ref = state_to_state[f"from_{i}"]["transition_dipole_moments"]
+            for ii, j in enumerate(range(i + 1, state.size)):
+                assert state.excitation_energy[j] == refevals[j]
+
+                # TODO Note the reversal
+                #      ... this is a bug in the testdata generator!
+                state2state = State2StateTransition(state, initial=j, final=i)
```

Comment:
Not too happy about the way you structure the tests here. Ideally I would construct only *one* `State2StateTransition` for each `i` and use that to test all `i -> j` tdms.

Reply:
Yes, I agree.

Reply:
Done.